api/resource.yaml

```diff
@@ -0,0 +1,2200 @@
+              example:
+                code: "RES-1003"
+                message: "Resource server not found"
+                description: "The resource server with the specified id does not exist"
+        "500":
+          description: Internal server error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                code: "RES-5000"
+                message: "Internal server error"
+                description: "An unexpected error occurred while processing the request"
+
+    delete:
+      tags:
+        - resource-servers
+      summary: Delete a resource server by id
+      description: Deletes a resource server. Cannot delete if it has resources or actions.
+      parameters:
+        - in: path
+          name: id
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        "204":
+          description: Resource server deleted
+        "400":
+          description: Bad request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              examples:
+                missing-id:
+                  summary: Missing resource server ID
+                  value:
+                    code: "RES-1002"
+                    message: "Invalid request format"
+                    description: "Resource server ID is required"
+                cannot-delete:
+                  summary: Cannot delete resource server
+                  value:
+                    code: "RES-1006"
+                    message: "Cannot delete resource server"
+                    description: "Cannot delete resource server that has resources or actions"
+        "404":
+          description: Resource server not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                code: "RES-1003"
+                message: "Resource server not found"
+                description: "The resource server with the specified id does not exist"
+        "500":
+          description: Internal server error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                code: "RES-5000"
+                message: "Internal server error"
+                description: "An unexpected error occurred while processing the request"
+
+  /resource-servers/{rsId}/resources:
+    get:
+      tags:
+        - resources
+      summary: List resources in resource server
+      description: |
+        Returns a paginated list of resources. Use the parentId parameter to filter:
+        - No parentId: returns all resources (flat list)
+        - parentId=null or empty: returns only top-level resources
```

Comment:
Shouldn't both of these (no parent id and parent id=nil) behave in the same manner? Feels not nice to change the behaviour based on a nil/ empty parameter.

Do we actually have a requirement to return all the resources including nested ones with this endpoint?

Reply:
removed